### PR TITLE
[Improve][flink-api]extraction unified method

### PR DESCRIPTION
**Pull request by liugddx**
## Purpose of this pull request

extraction unified method

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Review by EricJoy2048 — DISMISSED** [review]
+1

File: seatunnel-apis/seatunnel-api-flink/src/main/java/org/apache/seatunnel/flink/FlinkEnvironment.java [hunk 1]
```diff
@@ -322,4 +325,18 @@ private void setCheckpoint() {
         }
     }
 
+    public void registerResultTable(Config config, String tableName, DataStream<Row> dataStream) {
```
Comment (Hisoka-X) [inline seatunnel-apis/seatunnel-api-flink/src/main/java/org/apache/seatunnel/flink/FlinkEnvironment.java]: tableName always be `RESULT_TABLE_NAME`, so you can remove it.

Comment (liugddx) [reply seatunnel-apis/seatunnel-api-flink/src/main/java/org/apache/seatunnel/flink/FlinkEnvironment.java]: Done.

**Review by Hisoka-X — APPROVED** [review]
LGTM

**Review by EricJoy2048 — APPROVED** [review]
LGTM

**Comment by liugddx** [comment]
CI is error ,and i think it's because the `JdbcPostgresIT` and `JdbcMysqlIT` are unstable.and i tested it locally and the error also happen. so i fix it in #2907 .

**Comment by liugddx** [comment]
> CI is error ,and i think it's because the `JdbcPostgresIT` and `JdbcMysqlIT` are unstable.and i tested it locally and the error also happen. so i fix it in #2907 .

@EricJoy2048 please rerun CI,thanks. CC @ic4y

**Comment by EricJoy2048** [comment]
> #2907

Ok, I will rerun the failed job after the https://github.com/apache/incubator-seatunnel/pull/2907 been merged.

**Comment by liugddx** [comment]
Hi,please rerun CI. @EricJoy2048

**Comment by liugddx** [comment]
please rerun CI. @EricJoy2048

**Review by hailin0 — COMMENTED** [review]
LGTM

**Review by ashulin — APPROVED** [review]
LGTM

**Pull request merged**